tools/sftd_disco/sftd_disco.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+set -e
+exec 2>&1
+
+# Assumes /etc/wire/sftd-disco/ directory exists.
+
+USAGE="example usage: $0 _sft._tcp.wire-server-sftd.wire.svc.cluster.local"
+srv_name=${1?$USAGE}
+
+old="/etc/wire/sftd-disco/sft_servers_all.json"
+new="${old}.new"
+
+function valid_entry() {
+    local line=$1
+    # TODO sanity check that this is real dig output
+    return 0
+}
+
+function valid_url() {
+    local url=$1
+    #TODO basic sanity check
+    return 0
+}
+
+# for a given SRV record
+# 1. lookup the record
+# 2. for each entry: extract host and port and call 'curl host:port/sft/url'
+# 4. save the resulting URLs as a json array to a file
+# this file can then be served from nginx running besides sft
+function upstream() {
+    name=$1
+    port=${2:-'8585'}
+    entries=$(dig +short +retries=3 +search SRV ${name} | sort)
+    unset servers
+    comma="  "
+    IFS=$'\n'
+    for entry in ${entries[@]}; do
+        if valid_entry "$entry"; then
+            sft_host_port=$(echo "$entry" | awk '{print $4":"$3}')
+            sft_url=$(curl -s http://$sft_host_port/sft/url)
+            if valid_url "$sft_url"; then
+                servers+=("\n${comma}\"${sft_url}\"")
+                comma=", "
+            fi
+        fi
+    done
+    IFS=""
+    if [ -n "$servers" ]; then
+        printf "$(echo '{"sft_servers_all": [' ${servers[@]})\n]} \n" >>${new}
+    else
+        printf "" >>${new}
+    fi
+}
```

Comment:
Would you mind if we changed this such that the JSON object is exactly as requested in FS-266? That way I could reuse the JSON instances I already have and simply parse this without writing an additional parser.

In FS-266, each server item within the `sft_servers_all` object should be of the form: `{"urls": [$url_itself]}`.

Reply:
Never mind. The way it is now will be just fine.